pkg/repo/conversation_repost_histories/pg.go

```diff
@@ -0,0 +1,45 @@
+package conversation_repost_histories
+
+import (
+	"github.com/defipod/mochi/pkg/model"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type pg struct {
+	db *gorm.DB
+}
+
+func NewPG(db *gorm.DB) *pg {
+	return &pg{
+		db: db,
+	}
+}
+
+func (pg *pg) Upsert(model model.ConversationRepostHistories) error {
+	tx := pg.db.Begin()
+
+	err := tx.Clauses(clause.OnConflict{
+		Columns:   []clause.Column{{Name: "guild_id"}, {Name: "origin_channel_id"}, {Name: "origin_start_message_id"}},
+		UpdateAll: true,
+	}).Create(&model).Error
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	return tx.Commit().Error
+}
+
+func (pg *pg) Update(model *model.ConversationRepostHistories) error {
+	return pg.db.Save(model).Error
```

Comment:
`Save()` can create a new record and it doesn't match with the meaning of this function

Reply:
change to Save() with Where("id")